src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -120,9 +122,17 @@ public class GitLabServer extends AbstractDescribableImpl<GitLabServer> {
     private String hooksRootUrl;
 
     /**
-     * The secret token used while setting up hook url in the GitLab server
+     * The credentials id of the secret token used while setting up hook url in the GitLab server
      */
-    private Secret secretToken;
```

Comment:
This should be marked `transiant` and use a `readResolve` to transfer `secretToken` into the credentials id.

https://www.jenkins.io/doc/developer/persistence/backward-compatibility/

Otherwise, you would brick everyone's setup when upgrading to this plugin version.

You can see how this is implemented here:
https://github.com/jenkinsci/email-ext-plugin/blob/daea4f4780913aa0068ee842368c5f42bdfcbb68/src/main/java/hudson/plugins/emailext/MailAccount.java#L263-L268

Reply:
Good point. I will take a look at this.